src/libs/metadata_view.c

```diff
@@ -927,30 +1325,39 @@ static int lua_register_info(lua_State *L)
     lua_pop(L,1);
   }
   {
-    GtkLabel *name = GTK_LABEL(gtk_label_new(key));
-    GtkLabel *value = GTK_LABEL(gtk_label_new("-"));
-    gtk_widget_set_name(GTK_WIDGET(value), "brightbg");
-    gtk_label_set_selectable(value, TRUE);
-    gtk_widget_set_halign(GTK_WIDGET(name), GTK_ALIGN_START);
-    gtk_widget_set_halign(GTK_WIDGET(value), GTK_ALIGN_FILL);
-    gtk_label_set_xalign (value, 0.0f);
-    gtk_grid_attach_next_to(GTK_GRID(self->widget), GTK_WIDGET(name), NULL, GTK_POS_BOTTOM, 1, 1);
-    gtk_grid_attach_next_to(GTK_GRID(self->widget), GTK_WIDGET(value), GTK_WIDGET(name), GTK_POS_RIGHT, 1, 1);
-    gtk_widget_show_all(self->widget);
+    GtkTreeIter iter;
+    dt_lib_metadata_view_t *d = (dt_lib_metadata_view_t *)self->data;
+    gtk_list_store_append(GTK_LIST_STORE(d->model), &iter);
+    // get the row index
+    GtkTreePath *path = gtk_tree_model_get_path(d->model, &iter);
+    gint *i = gtk_tree_path_get_indices(path);
+    const int index = i[0];
+    gtk_tree_path_free(path);
+    gtk_list_store_set(GTK_LIST_STORE(d->model), &iter,
+                       DT_METADATA_COL_INDEX, index,
+                       DT_METADATA_COL_NAME, key,
+                       DT_METADATA_COL_NAME_L, key,
+                       DT_METADATA_COL_VALUE, "-",
+                       DT_METADATA_COL_VISIBLE, TRUE,
+                       DT_METADATA_COL_ORDER, index,
+                       -1);
     {
-      lua_getfield(L,-1,"widgets");
+      lua_getfield(L,-1,"indexes");
       lua_pushstring(L,key);
-      lua_pushlightuserdata(L,value);
+      lua_pushinteger(L,index);
       lua_settable(L,5);
       lua_pop(L,1);
     }
+    // apply again preferences because it's already done
```

Comment:
@wpferguson
Here preferences are called from lua_register_info(). I would fine more correct to call lua_register_info() from gui_init() at the right time (after or at the end of _lib_metadata_view_init_labels()).
Would that be correct ?
Should have to remove:

  lua_pushlightuserdata(L, self);
  lua_pushcclosure(L, lua_register_info,1);

from init() ?

Reply:
I'll look at it and see...

Reply:
lua_register_info() is a callback that runs when called from a script.  The scripts run long after gui_init() is done.  While I see the logic, I don't think it will work.  You could always try it and see what happens :smiley:

Reply:
>  While I see the logic, I don't think it will work

Thanks a lot for the advice ! :smiley: